backend/test/integration/openai.test.ts

```diff
@@ -176,22 +175,32 @@ describe('OpenAI Integration Tests', () => {
 		expect(chatResponse.completion?.content).toBe('Hi');
 		// check the chat history has been updated
 		expect(chatHistory.length).toBe(3);
-		// system role is added to the start of the chat history
-		expect(chatHistory[0].completion?.role).toBe('system');
-		expect(chatHistory[0].completion?.content).toBe(systemRoleDefault);
-		// rest of the chat history is in order
-		expect(chatHistory[1].completion?.role).toBe('user');
-		expect(chatHistory[1].completion?.content).toBe("I'm a user");
-		expect(chatHistory[2].completion?.role).toBe('assistant');
-		expect(chatHistory[2].completion?.content).toBe("I'm an assistant");
+		expect('completion' in chatHistory[0]);
+		expect('completion' in chatHistory[1]);
+		expect('completion' in chatHistory[2]);
+		if (
+			'completion' in chatHistory[0] &&
+			'completion' in chatHistory[1] &&
+			'completion' in chatHistory[2]
+		) {
+			// system role is added to the start of the chat history
+			expect(chatHistory[0].completion.role).toBe('system');
+			expect(chatHistory[0].completion.content).toBe(systemRoleDefault);
+
+			// rest of the chat history is in order
+			expect(chatHistory[1].completion.role).toBe('user');
+			expect(chatHistory[1].completion.content).toBe("I'm a user");
+			expect(chatHistory[2].completion.role).toBe('assistant');
+			expect(chatHistory[2].completion.content).toBe("I'm an assistant");
+		}
```

Comment:
I've had to restructure a bunch of the tests to look like this, because we cannot assume that any chatMessage will have a completion. I've yet to think of a more concise way to express the expect statements.

Reply:
Hmm. Can you not "coerce" into more specific type, e.g.

`const firstChatMessage = chatHistory[0] as ChatSystemMessage`?

Reply:
I've found it's nicer to just compare the whole chatHistory in one expect statement